src/PostgREST/Query/SqlFragment.hs

```diff
@@ -266,24 +280,37 @@ pgFmtOrderTerm qi ot =
     maybe mempty nullOrder $ otNullOrder ot])
   where
     fmtOTerm = \case
-      OrderTerm{otTerm}                        -> pgFmtField qi otTerm
-      OrderRelationTerm{otRelation, otRelTerm} -> pgFmtField (QualifiedIdentifier mempty otRelation) otRelTerm
+      OrderTerm{otTerm=(fn, jp)}                        -> pgFmtField qi (unknownField fn jp)
+      OrderRelationTerm{otRelation, otRelTerm=(fn, jp)} -> pgFmtField (QualifiedIdentifier mempty otRelation) (unknownField fn jp)
 
     direction OrderAsc  = "ASC"
     direction OrderDesc = "DESC"
 
     nullOrder OrderNullsFirst = "NULLS FIRST"
     nullOrder OrderNullsLast  = "NULLS LAST"
 
-
-pgFmtFilter :: QualifiedIdentifier -> Filter -> SQL.Snippet
-pgFmtFilter _ (FilterNullEmbed hasNot fld) = SQL.sql (pgFmtIdent fld) <> " IS " <> (if hasNot then "NOT" else mempty) <> " NULL"
-pgFmtFilter _ (Filter _ (NoOpExpr _)) = mempty -- TODO unreachable because NoOpExpr is filtered on QueryParams
-pgFmtFilter table (Filter fld (OpExpr hasNot oper)) = notOp <> " " <> case oper of
+-- | Interpret a literal in the way the planner indicated through the CoercibleField.
+pgFmtUnknownLiteralForField :: SQL.Snippet -> CoercibleField -> SQL.Snippet
+pgFmtUnknownLiteralForField value CoercibleField{cfTransform=(Just parserProc)} = pgFmtCallUnary parserProc value
+-- But when no transform is requested, we just use the literal as-is.
+pgFmtUnknownLiteralForField value _ = value
+
+-- | Array version of the above, used by ANY().
+pgFmtArrayLiteralForField :: [Text] -> CoercibleField -> SQL.Snippet
+-- When a transformation is requested, we need to apply the transformation to each element of the array. This could be done by just making a query with `parser(value)` for each value, but may lead to huge query lengths. Imagine `data_representations.color_from_text('...'::text)` for repeated for a hundred values. Instead we use `unnest()` to unpack a standard array literal and then apply the transformation to each element, like a map.
+-- Note the literals will be treated as text since in every case when we use ANY() the parameters are textual (coming from a query string). We want to rely on the `text->domain` parser to do the right thing.
+pgFmtArrayLiteralForField values CoercibleField{cfTransform=(Just parserProc)} = SQL.sql "(SELECT " <> pgFmtCallUnary parserProc (SQL.sql "unnest(" <> unknownLiteral (pgBuildArrayLiteral values) <> "::text[])") <> ")"
```

Comment:
Hm, would it be possible to not use the `::text[]` casting here? I'm a bit hesitant on having a hardcoded type on the code.

Reply:
This is not the only place we make this assumption. Anytime you pass a constant in a query parameter and we need to interpret it using data reps, we take it as being text (as opposed to an unknown type literal which functions can't accept).

Basically I think the only way we can _get_ a literal today is by someone specifying it as a filter using HTTP query parameters? I could be wrong. If I'm not though, the current setup makes sense to me. Since query parameters are always going to be text by definition, because that's what HTTP query parameters are, we need to clarify that to Postgres here so it can select the right data rep function overload. 

Like, there's no way to make a literal that's formed from JSON or something that our query parser and/or Postgres would be able to use. Am I wrong?

Reply:
>  Anytime you pass a constant in a query parameter and we need to interpret it using data reps, we take it as being text (as opposed to an unknown type literal which functions can't accept).

:thinking: Does this mean that we could change the `unknownLiteral` to use a `textEncoder` in all our filters and it would still work as usual?(all tests green?)

Reply:
I thought that was the case, but I'm still new to PostgREST, so I'm not sure. Where else would the literals come from when we generate the SQL, if not from the query string?

Reply:
> Does this mean that we could change the unknownLiteral to use a textEncoder in all our filters and it would still work as usual?(all tests green?)

Answering previous my question, it won't work because of:

```sql
select 1 = '1';
 ?column?
----------
 t
(1 row)


select 1 = '1'::text;
ERROR:  operator does not exist: integer = text
LINE 1: select 1 = '1'::text;
```